yaqd_andor/_andorsdk2_ixon.py

```diff
@@ -0,0 +1,277 @@
+        self.max_width = self._config["pixel_width"]
+        self.max_height = self._config["pixel_height"]
+        self.preamp_gain = int(self._config["simple_preamp_gain_control"])
+        self.has_mono = bool(self._config["has_monochromator"])
+        self._set_aoi()
+
+        if self.has_mono:
+            self.spec_position = self._state["spec_position"]
+            self.set_spec_position(self.spec_position)
+        else:
+            self._gen_mappings()
+
+        # implement config, state features
+        # see p97 of SDK2 on the conversion of noise filters from sdk3 to sdk2 or v-v.
+        # Commented out until further work done to establish differences between
+        # these filters and SDK3.
+        # self.sdk.Filter_SetMode(int(self._config["spurious_noise_filter"]))
+        # self.sdk.Filter_SetThreshold(float(self._config["filter_threshold"]))
+        self.sdk.SetPreAmpGain(self.preamp_gain)
+        self.logger.info(f"PreAmpGain: {self.preamp_gain}")
+
+        self.sdk.SetExposureTime(self.exposure_time)
+        self.logger.info(f"Exposure Time: {self.exposure_time} sec")
+        # aoi currently in config, so only need to run on startup
+
+        self._set_temperature()
+        self.sdk.SetShutter(int(0), int(1), int(100), int(100))
+
+    def _gen_mappings(self):
+        """Get map."""
+        if self.has_mono:
+            # translate inputs into appropriate internal units
+            spec_inclusion_angle_rad = np.radians(float(self._config["inclusion_angle"]))
+            spec_focal_length_tilt_rad = np.radians(float(self._config["focal_length_tilt"]))
+            pixel_width_mm = float(self.sensor_width / self.max_width) / 1000.00
+            self._channel_mappings = {"image": ["wavelengths", "y_index"]}
+            self._mapping_units = {"wavelengths": "nm", "y_index": "None"}
+            self.spec_position = self._state["spec_position"]
+
+            # create array
+            i_pixel = np.array(self.x_ai, dtype=float)
+            eff_pixel_width_mm = float(int(self.binning) * pixel_width_mm)
+            # calculate terms
+            x = np.arcsin(
+                (
+                    1e-6
+                    * self._config["order"]
+                    * self._config["grooves_per_mm"]
+                    * self.spec_position
+                )
+                / (2 * np.cos(spec_inclusion_angle_rad / 2.0))
+            )
+            A = np.sin(x - spec_inclusion_angle_rad / 2)
+            B = np.sin(
+                (spec_inclusion_angle_rad)
+                + x
+                - (spec_inclusion_angle_rad / 2)
+                - np.arctan(
+                    (
+                        eff_pixel_width_mm * (i_pixel - self._config["calibration_pixel"])
+                        + self._config["focal_length"] * spec_focal_length_tilt_rad
+                    )
+                    / (self._config["focal_length"] * np.cos(spec_focal_length_tilt_rad))
```

Comment:
I would contend that this mapping function does not work for a 2D camera...a tilt angle means that color changes with pixel height (i.e. the spectral mapping is 2D)

Reply:
Another consequence of tile is that there is no "calibration_pixel" but instead a no-vertical line of calibration pixels that follows the tilt angle.

Reply:
This implementation fails if tile angle is 90 (i.e. dispersion is along the opposite array axis).